src/components/Modal.js

```diff
@@ -77,17 +78,22 @@ const Modal = (props) => {
             <CustomStatusBar />
             <SafeAreaInsetsContext.Consumer>
                 {(insets) => {
-                    const {paddingTop, paddingBottom} = getSafeAreaPadding(insets);
+                    const {
+                        paddingTop: safeAreaPaddingTop,
+                        paddingBottom: safeAreaPaddingBottom,
+                    } = getSafeAreaPadding(insets);
+
                     return (
                         <View
                             style={{
-                                // This padding is based on the insets and could not neatly be
-                                // returned by getModalStyles to avoid passing this inline.
-                                paddingTop: needsSafeAreaPadding ? paddingTop : 20,
-
                                 ...styles.defaultModalContainer,
-                                paddingBottom,
                                 ...modalContainerStyle,
+                                paddingTop: shouldAddTopSafeAreaPadding
+                                    ? modalContainerStyle.paddingTop + safeAreaPaddingTop
+                                    : modalContainerStyle.paddingTop,
+                                paddingBottom: shouldAddBottomSafeAreaPadding
+                                    ? modalContainerStyle.paddingBottom + safeAreaPaddingBottom
+                                    : modalContainerStyle.paddingBottom,
```

Comment:
Not a blocker. Does this need to be above the `... modalContainerStyle`? I think we had moved that above so we could overwrite it with `modalContainerStyle`, but maybe that's no longer a problem now.

Reply:
That is no longer a problem, and it actually needs to be below `...modalContainerStyle` to work as intended now.
I wrote as that so we would be able to have the following cases:

- Modal style does not specifies `padding` or `shouldAdd***SafeAreaPadding` -> `padding = 0`;
- Modal style just specifies `shouldAdd***SafeAreaPadding` -> `padding = safeAreaPadding`
- Modal style specifies `padding` without `shouldAdd***SafeAreaPadding` -> `padding = specifiedPadding`;
- Modal style specifies both `padding` and `shouldAdd***SafeAreaPadding` -> `padding = sumOfBoth`.

Writing it below we can get all these use cases.

Reply:
That is no longer a problem, and it actually needs to be below `...modalContainerStyle` to work as intended now.
I wrote as that so we would be able to have the following cases:

- Modal style does not specifies `padding` or `shouldAdd***SafeAreaPadding` -> `padding = 0`;
- Modal style just specifies `shouldAdd***SafeAreaPadding` -> `padding = safeAreaPadding`
- Modal style specifies `padding` without `shouldAdd***SafeAreaPadding` -> `padding = specifiedPadding`;
- Modal style specifies both `padding` and `shouldAdd***SafeAreaPadding` -> `padding = sumOfBoth`.

Writing it below we can get all these use cases.